tests/functional/n/nan_comparison_check.txt

```diff
@@ -0,0 +1,12 @@
+nan-comparison:6::Comparison 'x is numpy.NaN' should be 'math.isnan(x)'
+nan-comparison:7::Comparison 'x == numpy.NaN' should be 'math.isnan(x)'
```

Comment:
why? to avoid importing the `math` package? in that case one could just do:
`from math import isnan`
I just don't trust as much the numpy implementation that could change in the future, as opposed to the math implementation which is just:

`def isnan(x):
    return x!=x`

@hippo91  what do you think?

Reply:
@orSolocate sorry i may have been unclear. What i wanted to say is that when people already use `numpy` such as in `x == numpy.NaN` then `x` is probably a `numpy` object and then IMHO it make more sense to recommend to use `numpy.isnan(x)`.
Of course when people are just dealing with regular float there is no reason to ask them using `numpy`.

Reply:
I decided not to make the fix. I would the `numpy` suggestion to be according to the name the user refers to `numpy` through. for example if the user uses `import numpy as np` i want to suggest to use `'np.isnan(x)'` . and I can't do that before you guys make your fix around the numpy library to include in your environment (the infer to numpy lib won't pass).
I decided to upload it as it is now (after rebase) and open a new feature to fix the test env, as well as to add this as an extended feature. sounds good? @hippo91